core/trino-parser/src/main/java/io/trino/sql/tree/SetTimeZone.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.sql.tree;
+
+import com.google.common.collect.ImmutableList;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static java.util.Objects.requireNonNull;
+
+public class SetTimeZone
+        extends Statement
+{
+    private final Optional<Expression> timeZone;
+    private final boolean local;
```

Comment:
IIUC, `local` is true if and only if `timeZone` is empty. 
Keeping `local` as a separate property creates risk of inconsistency. 
Do not keep this boolean but use `timeZone.isEmpty()` instead.